### PR TITLE
setup testing and add one very minimal test

### PR DESCRIPTION
**Pull request by jshipton**

### CODE REVIEW

**Comment by wence-** [comment]
I don't think you need the conftest.py file (it's sort of specific to the firedrake test suite).

File: tests/test_dcmip_setup.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    return stepper, dt
+
+def run_dcmip():
+
+    stepper, dt = setup_dcmip()
+    stepper.run(t=0, tmax=dt)
+
+def test_dcmip_runs():
+
+    run_dcmip()
```
Comment (wence-) [inline tests/test_dcmip_setup.py]: Can you assert something about the state of the system when this has run?  Or is it only the case that you're testing for the stepper actually running?

**Comment by wence-** [comment]
To wire this into travis, we'll need to install firedrake and then run the test suite in `.travis.yml`

File: tests/test_dcmip_setup.py (hunk 1 above)
Comment (jshipton) [inline tests/test_dcmip_setup.py]: Yes, I'm just testing that stepper runs - I really meant it when I said 'minimal'!

**Comment by jshipton** [comment]
OK, I have no clue how this travis testing works - shall I delete the contest.py file? What next?

**Comment by jshipton** [comment]
Thanks for looking at this Lawrence - I'm running for a train now but will carry on in the morning.

J


________________________________

File: tests/test_dcmip_setup.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
+from __future__ import absolute_import
+from dcore import *
+from firedrake import IcosahedralSphereMesh, ExtrudedMesh, Expression, \
+    VectorFunctionSpace
+import numpy as np
+
+def setup_dcmip():
+
+    nlayers = 2         # 10 horizontal layers
```
Comment (wence-) [inline tests/test_dcmip_setup.py]: This comment, it lies.

**Comment by wence-** [comment]
You could look at the thetis travis file https://github.com/thetisproject/thetis/blob/master/.travis.yml for an example.  Note how we set the PETSc options we're going to need.  I think you don't need to build with mumps for this.  I also do "caching" of the installed firedrake instance.  We can try without that bit to see if things take too long.

**Pull request merged**